docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -167,19 +173,36 @@ For more complex scenarios, such as a factory deploying multiple tokens with dif
 ----
 #[starknet::contract]
 mod MyToken {
+    use openzeppelin::token::erc20::ERC20 as erc20_component;
+    use openzeppelin::token::erc20::interface;
     use starknet::ContractAddress;
-    use openzeppelin::token::erc20::ERC20;
+
+    component!(path: erc20_component, storage: erc20, event: ERC20Event);
+
+    #[abi(embed_v0)]
+    impl ERC20Impl = erc20_component::ERC20Impl<ContractState>;
+    impl ERC20InternalImpl = erc20_component::InternalImpl<ContractState>;
 
     #[storage]
     struct Storage {
+        #[substorage(v0)]
+        erc20: erc20_component::Storage,
         // The decimals value is stored locally
-        _decimals: u8,
+        MyToken_decimals: u8
```

Comment:
why prefixing? nothing bad i guess, but that's for components -- as long as you're using components following the pattern

Reply:
Right, force of habit